src/chromatic_adaptation.rs

```diff
@@ -0,0 +1,285 @@
+                         ],
+                    inv_ma: [flt(1.8599364), flt(-1.1293816), flt(0.2198974),
+                             flt(0.3611914), flt(0.6388125), flt(-0.0000064),
+                             flt(0.0000000), flt(0.0000000), flt(1.0890636)
+                             ],
+                }
+            }
+             Method::XyzScaling => {
+                ConeResponseMatrices::<T> {
+                    ma: [flt(1.0000000), flt(0.0000000), flt(0.0000000),
+                         flt(0.0000000), flt(1.0000000), flt(0.0000000),
+                         flt(0.0000000), flt(0.0000000), flt(1.0000000)
+                         ],
+                    inv_ma: [flt(1.0000000), flt(0.0000000), flt(0.0000000),
+                             flt(0.0000000), flt(1.0000000), flt(0.0000000),
+                             flt(0.0000000), flt(0.0000000), flt(1.0000000)
+                             ],
+                }
+            }
+        }
+    }
+}
+
+
+///Converts color from one reference white point to another via chromatic adaptation (Swp -> Dwp)
+pub trait AdaptFrom<S, Swp, Dwp, T>: Sized
+    where T: Float,
+          Swp: WhitePoint<T>,
+          Dwp: WhitePoint<T>,
+          S: IntoColor<Swp, T>,
+{
+    ///Convert the source color to the destination color using the bradford method
+    fn adapt_from(color: S) -> Self;
+    ///Convert the source color to the destination color using the specified method
+    fn adapt_from_using<M: TransformMatrix<Swp, Dwp, T>>(color: S, method: M) -> Self;
+}
+
+impl<S, D, Swp, Dwp, T> AdaptFrom<S, Swp, Dwp, T> for D
+    where T: Float,
+          Swp: WhitePoint<T>,
+          Dwp: WhitePoint<T>,
+          S: IntoColor<Swp, T>,
+          D: FromColor<Dwp, T>,
+
+{
+    fn adapt_from(color: S) -> D {
+        Self::adapt_from_using(color, Method::Bradford)
+    }
+    fn adapt_from_using<M: TransformMatrix<Swp, Dwp, T>>(color:S, method: M) -> D {
+        let src_xyz: Xyz<Swp, T> = color.into_xyz();
+        let transform_matrix = method.generate_transform_matrix();
+        let arr = multiply_xyz(&transform_matrix, &src_xyz);
+        let dst_xyz: Xyz<Dwp, T> = Xyz::with_wp(arr[0], arr[1], arr[2]);
+        D::from_xyz(dst_xyz)
+    }
+}
+
+///Converts color from one reference white point to another using chromatic apaptation(Swp -> Dwp)
+pub trait AdaptInto<D, Swp, Dwp, T>: Sized
+    where T: Float,
+          Swp: WhitePoint<T>,
+          Dwp: WhitePoint<T>,
+{
+    ///Convert the source color to the destination color using the bradford method
+    fn adapt_into(self) -> D;
+    ///Convert the source color to the destination color using the specified method
+    fn adapt_into_using<M: TransformMatrix<Swp, Dwp, T>>(self, method: M) -> D;
+}
+
+impl<S, D, Swp, Dwp, T> AdaptInto<D, Swp, Dwp, T> for S
+    where T: Float,
+          Swp: WhitePoint<T>,
+          Dwp: WhitePoint<T>,
+          S: IntoColor<Swp, T>,
+          D: AdaptFrom<S, Swp, Dwp, T> + IntoColor<Dwp,T>
```

Comment:
Thinking about it, the only necessary requirement for `S` and `D` is that `D: AdaptFrom<S, Swp, Dwp, T>`, once the `S: IntoColor<Swp, T>` requirement has been removed from `AdaptFrom`.